app/decorators/event_generator_decorator.rb

```diff
@@ -0,0 +1,14 @@
+class EventGeneratorDecorator
```

Comment:
@andela-fsenjobi I don't see the need to create an `EventGeneratorDecorator` class. We already have an `EventDecorator` class so we can just use it. The event decorator can have 2 extra methods viz: `calendar` and `google_calendar_link`. calendar method can return an `Icalendar::Calendar` object. This will mean a lot of the boilerplate codes in the controller will be moved to the EventDecorator class. This is how we can use the the decorator in our controller.
```ruby
def generate
    calendar = @event.calendar
    headers['Content-Type'] = 'text/calendar; charset=UTF-8;'
    headers['Content-Disposition'] = 'attachment;' \
        "filename = #{@event.title.tr(' ', '_')}.ics"
    render text: calendar.to_ical
  end
```
And you won't need to create a `calendar_link` helper method. You can just use the one from the decorated event. So in your view, to add a google calendar link, you call `@event.google_calendar_link`.